### ISSUE TITLE
DigitalOcean droplet removal fails with "Timeout while waiting to for droplet to become 'off'

### ISSUE THREAD

**Issue by cmeisinger** [mitchellh/packer#3919]
- Packer version from `0.10.2/0.10.1`
- CentOS 6
- I can't grab output from the run due to https://github.com/mitchellh/packer/issues/2083 triggering another bug which causes the build to fail in a different way.

This seems to be the same bug as exhibited in https://github.com/mitchellh/packer/pull/3868, only in a different part of the process.  Unsure if this would be considered a duplicate or not.
# 

==> digitalocean: Gracefully shutting down droplet...
==> digitalocean: Error shutting down droplet: Timeout while waiting to for droplet to become 'off'
==> digitalocean: Destroying droplet...
==> digitalocean: Deleting temporary ssh key...
# Build 'digitalocean' errored: Error shutting down droplet: Timeout while waiting to for droplet to become 'off'

**Comment by rickard-von-essen** [comment]
Can you try from master? I did a fix for this I think. And at least at that
point building on DO worked fine.

**Comment by cmeisinger** [comment]
Build from master seems to have the same issue.  Here's the truncated output from the build log.

It looks like it tries to shut it down once, encounters the 422 and then SAYS it retries, however it just loops until it hits the timeout.

If I destroy the VM via the DigitalOcean web interface during this time, it removes itself without a problem.

```
2016/09/27 15:52:31 packer: 2016/09/27 15:52:31 [INFO] RPC client: Communicator ended with: 0
2016/09/27 15:52:31 ui: ==> digitalocean: Gracefully shutting down droplet...
==> digitalocean: Gracefully shutting down droplet...
2016/09/27 15:52:31 packer: 2016/09/27 15:52:31 Waiting for up to 600 seconds for droplet to become off
2016/09/27 15:52:31 packer: 2016/09/27 15:52:31 Checking droplet status... (attempt: 1)
2016/09/27 15:52:31 packer: 2016/09/27 15:52:31 ShutdownDroplet attempt #2...
2016/09/27 15:52:31 packer: 2016/09/27 15:52:31 Shutdown retry error: POST https://api.digitalocean.com/v2/droplets/27127146/actions: 422 there are currently blocking events processing for one or more of the droplets
2016/09/27 15:52:34 packer: 2016/09/27 15:52:34 Checking droplet status... (attempt: 2)
2016/09/27 15:52:37 packer: 2016/09/27 15:52:37 Checking droplet status... (attempt: 3)
2016/09/27 15:52:41 packer: 2016/09/27 15:52:41 Checking droplet status... (attempt: 4)
2016/09/27 15:52:44 packer: 2016/09/27 15:52:44 Checking droplet status... (attempt: 5)
2016/09/27 15:52:47 packer: 2016/09/27 15:52:47 Checking droplet status... (attempt: 6)
2016/09/27 15:52:50 packer: 2016/09/27 15:52:50 Checking droplet status... (attempt: 7)
2016/09/27 15:52:51 packer: 2016/09/27 15:52:51 ShutdownDroplet attempt #3...
2016/09/27 15:52:53 packer: 2016/09/27 15:52:53 Checking droplet status... (attempt: 8)
2016/09/27 15:52:56 packer: 2016/09/27 15:52:56 Checking droplet status... (attempt: 9)
2016/09/27 15:53:00 packer: 2016/09/27 15:53:00 Checking droplet status... (attempt: 10)
2016/09/27 15:53:03 packer: 2016/09/27 15:53:03 Checking droplet status... (attempt: 11)
2016/09/27 15:53:06 packer: 2016/09/27 15:53:06 Checking droplet status... (attempt: 12)
2016/09/27 15:53:09 packer: 2016/09/27 15:53:09 Checking droplet status... (attempt: 13)
2016/09/27 15:53:11 packer: 2016/09/27 15:53:11 ShutdownDroplet attempt #4...
2016/09/27 15:53:12 packer: 2016/09/27 15:53:12 Checking droplet status... (attempt: 14)
2016/09/27 15:53:16 packer: 2016/09/27 15:53:16 Checking droplet status... (attempt: 15)

2016/09/27 15:53:19 packer: 2016/09/27 15:53:19 Checking droplet status... (attempt: 16)
2016/09/27 15:53:22 packer: 2016/09/27 15:53:22 Checking droplet status... (attempt: 17)
2016/09/27 15:53:25 packer: 2016/09/27 15:53:25 Checking droplet status... (attempt: 18)
2016/09/27 15:53:29 packer: 2016/09/27 15:53:29 Checking droplet status... (attempt: 19)
2016/09/27 15:53:31 packer: 2016/09/27 15:53:31 ShutdownDroplet attempt #5...
2016/09/27 15:53:32 packer: 2016/09/27 15:53:32 Checking droplet status... (attempt: 20)
2016/09/27 15:53:35 packer: 2016/09/27 15:53:35 Checking droplet status... (attempt: 21)
2016/09/27 15:53:38 packer: 2016/09/27 15:53:38 Checking droplet status... (attempt: 22)
2016/09/27 15:53:41 packer: 2016/09/27 15:53:41 Checking droplet status... (attempt: 23)
2016/09/27 15:53:45 packer: 2016/09/27 15:53:45 Checking droplet status... (attempt: 24)
2016/09/27 15:53:48 packer: 2016/09/27 15:53:48 Checking droplet status... (attempt: 25)
2016/09/27 15:53:51 packer: 2016/09/27 15:53:51 Checking droplet status... (attempt: 26)
2016/09/27 15:53:51 packer: 2016/09/27 15:53:51 ShutdownDroplet attempt #6...
2016/09/27 15:53:54 packer: 2016/09/27 15:53:54 Checking droplet status... (attempt: 27)
2016/09/27 15:53:57 packer: 2016/09/27 15:53:57 Checking droplet status... (attempt: 28)
2016/09/27 15:54:01 packer: 2016/09/27 15:54:01 Checking droplet status... (attempt: 29)
2016/09/27 15:54:04 packer: 2016/09/27 15:54:04 Checking droplet status... (attempt: 30)
2016/09/27 15:54:07 packer: 2016/09/27 15:54:07 Checking droplet status... (attempt: 31)
2016/09/27 15:54:10 packer: 2016/09/27 15:54:10 Checking droplet status... (attempt: 32)
2016/09/27 15:54:12 packer: 2016/09/27 15:54:12 ShutdownDroplet attempt #7...
2016/09/27 15:54:13 packer: 2016/09/27 15:54:13 Checking droplet status... (attempt: 33)
2016/09/27 15:54:17 packer: 2016/09/27 15:54:17 Checking droplet status... (attempt: 34)
2016/09/27 15:54:20 packer: 2016/09/27 15:54:20 Checking droplet status... (attempt: 35)
2016/09/27 15:54:23 packer: 2016/09/27 15:54:23 Checking droplet status... (attempt: 36)
2016/09/27 15:54:26 packer: 2016/09/27 15:54:26 Checking droplet status... (attempt: 37)
2016/09/27 15:54:29 packer: 2016/09/27 15:54:29 Checking droplet status... (attempt: 38)
2016/09/27 15:54:32 packer: 2016/09/27 15:54:32 ShutdownDroplet attempt #8...
2016/09/27 15:54:33 packer: 2016/09/27 15:54:33 Checking droplet status... (attempt: 39)
2016/09/27 15:54:36 packer: 2016/09/27 15:54:36 Checking droplet status... (attempt: 40)
2016/09/27 15:54:39 packer: 2016/09/27 15:54:39 Checking droplet status... (attempt: 41)
2016/09/27 15:54:42 packer: 2016/09/27 15:54:42 Checking droplet status... (attempt: 42)
2016/09/27 15:54:45 packer: 2016/09/27 15:54:45 Checking droplet status... (attempt: 43)
2016/09/27 15:54:49 packer: 2016/09/27 15:54:49 Checking droplet status... (attempt: 44)
...
...
2016/09/27 16:02:16 packer: 2016/09/27 16:02:16 ShutdownDroplet attempt #31...
2016/09/27 16:02:19 packer: 2016/09/27 16:02:19 Checking droplet status... (attempt: 184)
2016/09/27 16:02:22 packer: 2016/09/27 16:02:22 Checking droplet status... (attempt: 185)
2016/09/27 16:02:25 packer: 2016/09/27 16:02:25 Checking droplet status... (attempt: 186)
2016/09/27 16:02:28 packer: 2016/09/27 16:02:28 Checking droplet status... (attempt: 187)
2016/09/27 16:02:31 ui error: ==> digitalocean: Error shutting down droplet: Timeout while waiting to for droplet to become 'off'
2016/09/27 16:02:31 ui: ==> digitalocean: Destroying droplet...
==> digitalocean: Error shutting down droplet: Timeout while waiting to for droplet to become 'off'
==> digitalocean: Destroying droplet...
2016/09/27 16:02:31 ui: ==> digitalocean: Deleting temporary ssh key...
==> digitalocean: Deleting temporary ssh key...
2016/09/27 16:02:31 ui error: Build 'digitalocean' errored: Error shutting down droplet: Timeout while waiting to for droplet to become 'off'
2016/09/27 16:02:31 Builds completed. Waiting on interrupt barrier...
2016/09/27 16:02:31 machine readable: error-count []string{"1"}
2016/09/27 16:02:31 ui error: 
==> Some builds didn't complete successfully and had errors:
2016/09/27 16:02:31 machine readable: digitalocean,error []string{"Error shutting down droplet: Timeout while waiting to for droplet to become 'off'"}
2016/09/27 16:02:31 ui error: --> digitalocean: Error shutting down droplet: Timeout while waiting to for droplet to become 'off'
2016/09/27 16:02:31 ui: 
==> Builds finished but no artifacts were created.
2016/09/27 16:02:31 waiting for all plugin processes to complete...
Build 'digitalocean' errored: Error shutting down droplet: Timeout while waiting to for droplet to become 'off'

==> Some builds didn't complete successfully and had errors:
--> digitalocean: Error shutting down droplet: Timeout while waiting to for droplet to become 'off'

==> Builds finished but no artifacts were created.
2016/09/27 16:02:31 /opt/packer-master/packer: plugin process exited
2016/09/27 16:02:31 /opt/packer-master/packer: plugin process exited
```

**Comment by bensquire** [comment]
Same issue here, the VM seems to work fine though...

**Comment by bahchis** [comment]
I have hit the same issue when using Packer 0.10.2 too.
Were you able to make any progress on this?

So far I was able to work around it with adding shell provisioner at the end which will shut down the machine in the background a minute after it is executed:
```
{
    "type": "shell",
    "inline": [
      "shutdown -h 1 &"
  ]
}
```

This at least creates the snapshot but the packer build fails with the following message ( probably #3853 ):
`==> digitalocean: Error waiting for snapshot to complete: Timeout while waiting to for droplet to become 'active`'

**Comment by rickard-von-essen** [comment]
Can anyone provide some more info:
1) How often do this happen? Every time, ever second. etc.
2) Can you share a simple template where you got this error? 

I would like to squash this bug quickly.

**Comment by bensquire** [comment]
I am building an image using shell provisioners (quite complex) and it happens every time. Eventually packer times out and disconnects. I'll see if I can throw a simple script together.

**Comment by bensquire** [comment]
@rickard-von-essen Failed with no provisioners as well...
<img width="764" alt="screen shot 2016-10-08 at 22 57 46" src="https://cloud.githubusercontent.com/assets/600237/19216206/bcd9dafa-8daa-11e6-9e09-d999372c04a3.png">

**Comment by rickard-von-essen** [comment]
Tried the above example 5 times and had no problems. Read the code and it looks correct.

Maybe you can build from master and apply this patch to get some more debug info:

```patch
diff --git a/builder/digitalocean/wait.go b/builder/digitalocean/wait.go
index 3544667..7b0e236 100644
--- a/builder/digitalocean/wait.go
+++ b/builder/digitalocean/wait.go
@@ -71,13 +71,15 @@ func waitForDropletState(
                for {
                        attempts += 1

-                       log.Printf("Checking droplet status... (attempt: %d)", attempts)
                        droplet, _, err := client.Droplets.Get(dropletId)
                        if err != nil {
                                result <- err
                                return
                        }

+                       log.Printf("Checking droplet status... (attempt: %d) state: %s",
+                               attempts, droplet.Status)
+
                        if droplet.Status == desiredState {
                                result <- nil
                                return
@@ -121,13 +123,14 @@ func waitForActionState(
                for {
                        attempts += 1

-                       log.Printf("Checking action status... (attempt: %d)", attempts)
                        action, _, err := client.DropletActions.Get(dropletId, actionId)
                        if err != nil {
                                result <- err
                                return
                        }

+                       log.Printf("Checking action status... (attempt: %d) status: %s",
+                               attempts, action.Status)
                        if action.Status == desiredState {
                                result <- nil
                                return
```

**Comment by onorua** [comment]
I've also faced with this nasty bug around 0.10.1, but I do confirm that version `0.11.1` (current master) doesn't have this bug anymore.

**Comment by rickard-von-essen** [comment]
I'll close this unless someone can confirm that they still sees this with 0.11.0.

**Comment by cmeisinger** [comment]
Please close.  0.11.0 gets through this error, and generates a different one now.

**Comment by potato20** [comment]
Still happening on 0.12.3. Just trying to create a debian-based image, all I did was install make and gcc.

**Comment by rickard-von-essen** [comment]
@potato20 open a new issue
